src/Microsoft.NetCore.Analyzers/Core/Security/ApprovedCipherModeAnalyzer.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Security
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    class ApprovedCipherModeAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string DiagnosticId = "CA5358";
+        private static readonly LocalizableString Title = new LocalizableResourceString(
+            nameof(SystemSecurityCryptographyResources.ApprovedCipherMode),
+            SystemSecurityCryptographyResources.ResourceManager,
+            typeof(SystemSecurityCryptographyResources));
+        private static readonly LocalizableString Message = new LocalizableResourceString(
+            nameof(SystemSecurityCryptographyResources.ApprovedCipherModeMessage),
+            SystemSecurityCryptographyResources.ResourceManager,
+            typeof(SystemSecurityCryptographyResources));
+        private static readonly LocalizableString Description = new LocalizableResourceString(
+            nameof(SystemSecurityCryptographyResources.ApprovedCipherModeDescription),
+            SystemSecurityCryptographyResources.ResourceManager,
+            typeof(SystemSecurityCryptographyResources));
+
+        internal static DiagnosticDescriptor Rule =
+            CreateDiagnosticDescriptor(DiagnosticId, Title, Message, Description);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        internal static string CipherModeTypeMetadataName =>
```

Comment:
You could remove this variable entirely, and directly use `WellKnownTypes.SystemSecurityCryptographyCipherMode` below.  I'll merge as is, and if you want, you can remove it in a future PR.

`DoNotUseInsecureDeserializerBinaryFormatterMethods` had an instance variable for the type, just so that the base class could use it.